include/openthread/commissioner.h

```diff
@@ -192,6 +192,25 @@ otError otCommissionerStart(otInstance *                 aInstance,
                             otCommissionerJoinerCallback aJoinerCallback,
                             void *                       aCallbackContext);
 
+/**
+ * This function enables the Thread Commissioner role with the supplied ID.
+ *
+ * @param[in]  aInstance         A pointer to an OpenThread instance.
+ * @param[in]  aId               A pointer to a Commissioner ID.
+ * @param[in]  aStateCallback    A pointer to a function that is called when the commissioner state changes.
+ * @param[in]  aJoinerCallback   A pointer to a function that is called with a joiner event occurs.
+ * @param[in]  aCallbackContext  A pointer to application-specific context.
+ *
+ * @retval OT_ERROR_NONE           Successfully started the Commissioner role.
+ * @retval OT_ERROR_INVALID_STATE  Commissioner is already started.
+ *
+ */
+otError otCommissionerStartWithId(otInstance *                 aInstance,
+                                  const char *                 aId,
+                                  otCommissionerStateCallback  aStateCallback,
+                                  otCommissionerJoinerCallback aJoinerCallback,
+                                  void *                       aCallbackContext);
```

Comment:
How about adding an API to set the ID (something like `otCommissionerSetId(const char *aId)`)?
We can allow it to be set when commissioner is not yet started (return error otherwise).
We can also then add a "get API" as well to get the currently set ID.

I think it may also map easier to `cli` command  (we can add `commissioner id` sub-command).

Reply:
I think it's a good idea. It's more elegant than my previous approach.